lib/src/circular_slider_paint.dart

```diff
@@ -1,34 +1,54 @@
 import 'package:flutter/gestures.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_circular_slider/src/base_painter.dart';
-import 'package:flutter_circular_slider/src/slider_painter.dart';
-import 'package:flutter_circular_slider/src/utils.dart';
+
+import 'base_painter.dart';
+import 'slider_painter.dart';
+import 'utils.dart';
+
+enum CircularSliderMode { singleHandler, doubleHandler }
+
+enum SlidingState { none, endIsBiggerThanStart, endIsSmallerThanStart }
+
+typedef SelectionChanged<T> = void Function(T a, T b, T c);
 
 class CircularSliderPaint extends StatefulWidget {
+  final CircularSliderMode mode;
   final int init;
   final int end;
-  final int intervals;
+  final int divisions;
   final int primarySectors;
   final int secondarySectors;
-  final Function onSelectionChange;
+  final SelectionChanged<int> onSelectionChange;
+  final Function onSelectionEnd;
```

Comment:
we could use the same typedef that I created for `onSelectionChange` (see line above)

Reply:
please note that we should also return the laps now